R/time_bins.R

```diff
@@ -130,68 +157,92 @@ time_bins <- function(interval = "Phanerozoic", rank = "stage",
+    }
+    if (is.character(interval) && length(interval) == 2) {
+      # rank ages
+      rank_ages <- df[which(df$rank == rank),]
+      w <- which(df$interval_name %in% interval)
+      if (length(w) != length(interval)) {
+        stop(
+          paste0(
+            "Check spelling of specified intervals.
   Available intervals are accessible via GTS2020 and GTS2012."
-            )
           )
-        }
-        rank_ages <-
-          rank_ages[which(rank_ages$max_ma > min(df$min_ma[w]) &
-                            rank_ages$min_ma < max(df$max_ma[w])), ]
-        df <- subset(df, max_ma <= max(rank_ages$max_ma))
-        df <- subset(df, min_ma >= min(rank_ages$min_ma))
+        )
       }
+      rank_ages <-
+        rank_ages[which(rank_ages$max_ma > min(df$min_ma[w]) &
+                          rank_ages$min_ma < max(df$max_ma[w])),]
+      df <- subset(df, max_ma <= max(rank_ages$max_ma))
+      df <- subset(df, min_ma >= min(rank_ages$min_ma))
+    }
 
     #subset to rank
-    df <- df[which(df$rank == rank), ]
+    df <- df[which(df$rank == rank),]
 
     #numeric ages entered
     if (is.numeric(interval) && length(interval) == 1) {
-        if (interval > max(df$max_ma) || interval < min(df$min_ma)) {
-          stop("Value does not appear in the range of available intervals:
+      if (interval > max(df$max_ma) || interval < min(df$min_ma)) {
+        stop("Value does not appear in the range of available intervals:
     0 to 541")
-        }
-        int_index <-
-          which(interval <= df$max_ma & interval >= df$min_ma)
-        df <- df[int_index, ]
       }
+      int_index <-
+        which(interval <= df$max_ma & interval >= df$min_ma)
+      df <- df[int_index,]
+    }
 
-      if (is.numeric(interval) && length(interval) == 2) {
-        max_int <- max(interval)
-        min_int <- min(interval)
+    if (is.numeric(interval) && length(interval) == 2) {
+      max_int <- max(interval)
+      min_int <- min(interval)
 
-        if (max_int > max(df$max_ma) || min_int < min(df$min_ma)) {
-          stop("Values do not appear in the range of available intervals:
+      if (max_int > max(df$max_ma) || min_int < min(df$min_ma)) {
+        stop("Values do not appear in the range of available intervals:
           0 to 541")
-        }
-
-        int_index <-
-          which(min_int <= df$max_ma & max_int >= df$min_ma)
-        df <- df[int_index, ]
       }
 
+      int_index <-
+        which(min_int <= df$max_ma & max_int >= df$min_ma)
+      df <- df[int_index,]
+    }
+  } else {
+    # Assign input scale to df
+    df <- scale
+    # Update column names to work with getScaleData
+    colnames(df)[which(colnames(df) %in% c("name",
+                                           "max_age",
+                                           "min_age"))] <- c("interval_name",
+                                                             "max_ma",
+                                                             "min_ma")
```

Comment:
This assumes the data.frame comes from getScaleData and the columns are always in this order. The documentation implies that user-made data.frames can also have these column names, but they might not necessarily be in this order. Should we modify this bit of code to accommodate this?

Reply:
Definitely, I've now implemented a fix for this. Let me know what you think!